pkg/api/kubernetesagentpool/types.go

```diff
@@ -0,0 +1,114 @@
+package kubernetesagentpool
+
+const (
+	APIVersion = "kubernetesagentpool"
+)
+
+type AgentPool struct {
+	ID       string                `json:"id,omitempty"`
+	Location string                `json:"location,omitempty"`
+	Name     string                `json:"name,omitempty"`
+	Plan     *ResourcePurchasePlan `json:"plan,omitempty"`
+	Tags     map[string]string     `json:"tags,omitempty"`
+	Type     string                `json:"type,omitempty"`
+
+	Properties *Properties `json:"properties"`
+}
+
+type Properties struct {
+	KubernetesVersion       string                   `json:"kubernetesVersion"`
+	KubernetesEndpoint      string                   `json:"kubernetesEndpoint"`
+	DnsPrefix               string                   `json:"dnsPrefix,omitempty"`
+	Version                 string                   `json:"version,omitempty"`
+	AgentPoolProfiles       []*AgentPoolProfile      `json:"agentPoolProfiles,omitempty"`
+	WindowsProfile          *WindowsProfile          `json:"windowsProfile,omitempty"`
+	LinuxProfile            *LinuxProfile            `json:"linuxProfile,omitempty"`
+	NetworkProfile          *NetworkProfile          `json:"networkProfile,omitempty"`
+	JumpBoxProfile          *JumpBoxProfile          `json:"jumpboxProfile,omitempty"`
+	ServicePrincipalProfile *ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
+	CertificateProfile      *CertificateProfile      `json:"certificateProfile,omitempty"`
+}
+
+type CertificateProfile struct {
+	// CaCertificate is the certificate authority certificate.
+	CaCertificate string `json:"caCertificate,omitempty"`
+	// CaPrivateKey is the certificate authority key.
+	CaPrivateKey string `json:"caPrivateKey,omitempty"`
+	// ApiServerCertificate is the rest api server certificate, and signed by the CA
+	APIServerCertificate string `json:"apiServerCertificate,omitempty"`
+	// ApiServerPrivateKey is the rest api server private key, and signed by the CA
+	APIServerPrivateKey string `json:"apiServerPrivateKey,omitempty"`
+	// ClientCertificate is the certificate used by the client kubelet services and signed by the CA
+	ClientCertificate string `json:"clientCertificate,omitempty"`
+	// ClientPrivateKey is the private key used by the client kubelet services and signed by the CA
+	ClientPrivateKey string `json:"clientPrivateKey,omitempty"`
+	// KubeConfigCertificate is the client certificate used for kubectl cli and signed by the CA
+	KubeConfigCertificate string `json:"kubeConfigCertificate,omitempty"`
+	// KubeConfigPrivateKey is the client private key used for kubectl cli and signed by the CA
+	KubeConfigPrivateKey string `json:"kubeConfigPrivateKey,omitempty"`
+}
+
+type ServicePrincipalProfile struct {
+	ClientID string `json:"servicePrincipalClientID,omitempty"`
+	Secret   string `json:"servicePrincipalClientSecret,omitempty"`
+}
+
+type JumpBoxProfile struct {
```

Comment:
So while some of this is similar, it's use case is called from a very unique point during an HTTPs transaction with the ACS RP.. which means (if you look closely) there are a number of differences between the structs defined here, and the structs defined in the "public" API model..

For instance:
1. Kubernetes version
2. Kubernetes endpoints
3. TLS Certificate material

Are all unique to this request.. The underlying code that parses the new API was deliberately chosen to uncouple itself from the concept of a `ContainerService{}` as the use case here is different (we aren't creating a container service)

See my notes https://github.com/Azure/acs-engine/pull/957#issuecomment-318453486 that explain why we decided to do things this way.

TLDR; It's similar code, but it's not duplicating anything